### PR TITLE
fix nil pointer deref when trying to log an error in docker-up

### LINKED ISSUE #7983: Workspace shutdown unexpectedly
### Bug description

On February 1 @ 11:02pm UTC, the workspace shutdown unexpectedly after a minute.

[Related discord thread.](https://discord.com/channels/816244985187008514/938192626543362140)

### Steps to reproduce

@shaal are you able to recreate this with any particular public repos?

### Workspace affected

https://phase2-hfdrupal-s60ok2xajnq.ws-us30.gitpod.io/

### Expected behavior

The workspace does not unexpectly stop a minute after starting.

### Example repository

n/a

### Anything else?

[The related logs](https://cloudlogging.app.goo.gl/aJwijQ9mJ1Hvn86k6)

### PR DESCRIPTION
**Pull request by sagor999**
## Description

I think there is a nil pointer deref in here since `log` is not initialized. 
https://github.com/gitpod-io/gitpod/blob/5b74af751837f95839e0a645b42bbdd10748fc40/components/docker-up/docker-up/main.go#L172

## Related Issue(s)

Fixes #7983

## Release Notes

```release-note
none
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **sagor999** after the PR has been reviewed.

Associated issue: *#7983*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/docker-up/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/docker-up/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by csweichel — APPROVED** [review]
LGTM

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 8b0b784738a34dab78504c354eb8dd5c70f91063</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7989#pullrequestreview-872196031" title="Approved">csweichel</a>*

Associated issue: *#7983*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/docker-up/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/docker-up/OWNERS)~~ [csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
